index.html

```diff
@@ -2138,6 +2139,22 @@ <h2>
                 </li>
               </ol>
             </li>
+            <li>If <var>details</var>["<a>regionCode</a>"] is present and not
+            the empty string:
+              <ol>
+                <li>Let <var>regionCode</var> be the result of <a>strip leading
+                and trailing ASCII whitespace</a> and <a data-cite=
+                "!INFRA#ascii-uppercase">ASCII uppercasing</a>
+                <var>details</var>["<a>regionCode</a>"].
+                </li>
+                <li>If <var>regionCode</var> is not a valid [[!ISO3166-2]]
+                subdivision code, throw a <a>RangeError</a> exception.
+                </li>
+                <li>Set <var>address</var>.<a>[[\regionCode]]</a> to
+                <var>regionCode</var>.
+                </li>
+              </ol>
+            </li>
```

Comment:
Should we consider deriving region from regionCode, so the caller only has to pass one? I'm thinking ahead to the constructor; do we want to prevent `new PaymentAddress({ ..., regionCode: "PT-11", region: "Tasmania" })`?

I guess for the cases where there is no corresponding regionCode, we'd still want to pass region. But if regionCode is not the empty string, deriving region from it might be good...

Reply:
> But if regionCode is not the empty string, deriving region from it might be good...

Agree. It's definitely doable. ISO3166-2 provides lookup tables, for example:

```
AD-07* Andorra la Vella
AD-02* Canillo
AD-03* Encamp
AD-08* Escaldes-Engordany
AD-04* La Massana
AD-05* Ordino
AD-06* Sant Julià de Lòria
```

Where **column 2** (defined thing) is "...the country subdivision names in the administrative language of the country concerned, where relevant with diacritic signs..." (as Unicode). Note that "a country’s administrative language is a written language used by the administration of the country at the national level". So, the result won't be in English a lot of the time - but that's fine, IMO. This is not for display purposes. 

So, something like:

The steps to derive a region from a country subdivision code element are as follows. The string takes a DOMString <var>subCode</var> as input and returns a DOMString.  
 
 1. If validate a region code <var>subCode</var> return false, throw `RangeError` exception.
 1. ASCII uppercase <var>subCode</var>, and let <var>normalizeCode</var> be the result.
 1. Find <var>normalizeCode</var> in <cite>Section 8, List of country subdivision names and their code elements</cite> and return the value of "column 2" for the matching country subdivision code element. 

Sound good?

Reply:
That sounds great! Probably we should get implementer buy-ins though; maybe as a separate PR? Because it'd mean having to ship that table with the browser...

Reply:
Agree... ☝️@rsolomakhin, @mnoorenberghe, something to start thinking about. Would appreciate your input.

Reply:
We are already using libaddressinput (and I believe Chromium does too) so it would be great if we could also use that here instead of having two versions of similar data. Does ISO3166-2 only have one name per region? What about when there are multiple official languages? Consider CA-QC: is it "[Quebec](https://chromium-i18n.appspot.com/ssl-address/data/CA)" or "[Québec](https://chromium-i18n.appspot.com/ssl-address/data/CA--fr)" in ISO3166-2. libaddressinput provides both but without a clear way to know which one would match ISO3166-2.

Reply:
So, just to troll us, the spec is in English and French side-by-side 👍. Presumedly, we can just grab the "en" column 2: 

![screenshot 2018-03-16 16 13 35](https://user-images.githubusercontent.com/870154/37504829-1bf014a6-2935-11e8-87c3-1466b7131e22.png)

Reply:
> Presumedly, we can just grab the "en" column 2

I'm not quite sure how this all fits together but won't this affect what gets returned in the `region` property and therefore may be displayed back to the user on the payment confirmation page the merchant renders? If so, it would seem like we should use the language best for the user.

Reply:
We can probably perform some kind of lookup, based on the document language. But then it gets into preference order for when it doesn't match the user's language (and there is no "en").

Reply:
Seems crazy town, but: 

   1. Try to match document language. 
   1. Try to match user language (like Accept-Language)? 
   1. Try to match "en".
   1. Um... just give me whatever you got.

Reply:
We use libaddressinput in Chromium and @sebsg is helping us to add the ISO codes to that.

Reply:
> But if regionCode is not the empty string, deriving region from it might be good.

The `regionCode` field takes care of interop, so we don't necessarily have to agree on how the `region` field is derived from that. Having said that, the rule of thumb we have been using in Chrome is to use the browser user interface language when deciding between Quebec and Québec. If the browser user interface language does not match any of the languages supported by the country, then Chrome falls back to the first language specified in libaddressinput data for that country. For example, if the browser user interface language is Italian, then Canadian addresses will be in English. Although the default language works out to be English for Canada, that's not something that we should hardcode for all countries in the world, IMHO.